src/v0/destinations/klaviyo/util.js

```diff
@@ -135,6 +140,7 @@ const createCustomerProperties = (message) => {
     message,
     MAPPING_CONFIG[CONFIG_CATEGORIES.PROFILE.name],
   );
+  customerProperties.$id = getFieldValueFromMessage(message, 'userId');
```

Comment:
check on whether `userId` is null not required to avoid setting null to `$id` ?

Reply:
In the next line, I am using the `removeUndefinedAndNullValues` for the same.